pkg/datapath/linux/probes/probes.go

```diff
@@ -156,7 +161,7 @@ func (p *ProbeManager) SystemConfigProbes() error {
 	config := p.features.SystemConfig
 	// Required
 	if !config.ConfigBpf.Enabled() {
-		return fmt.Errorf("CONFIG_BPF kernel parameter is required")
+		return ErrKernelConfigNotFound
```

Comment:
Can we actually detect that kernel config is not found? Right now we are guessing based whether `CONFIG_BPF` was set. Imagine a situation in which kernel config was available, but `CONFIG_BPF` was not set.

Reply:
Unfortunately bpftool with option `-j` doesn't show any warning when the kernel config is not available, but it rather sets everything to `null`. This is the output:

```
❯ sudo ./bpftool feature probe -j | jq '.system_config'
{
  "unprivileged_bpf_disabled": 1,
  "bpf_jit_enable": 1,
  "bpf_jit_harden": 0,
  "bpf_jit_kallsyms": 0,
  "bpf_jit_limit": 264241152,
  "CONFIG_BPF": null,
  "CONFIG_BPF_SYSCALL": null,
  "CONFIG_HAVE_EBPF_JIT": null,
  "CONFIG_BPF_JIT": null,
  "CONFIG_BPF_JIT_ALWAYS_ON": null,
  "CONFIG_CGROUPS": null,
  "CONFIG_CGROUP_BPF": null,
  "CONFIG_CGROUP_NET_CLASSID": null,
  "CONFIG_SOCK_CGROUP_DATA": null,
  "CONFIG_BPF_EVENTS": null,
  "CONFIG_KPROBE_EVENTS": null,
  "CONFIG_UPROBE_EVENTS": null,
  "CONFIG_TRACING": null,
  "CONFIG_FTRACE_SYSCALLS": null,
  "CONFIG_FUNCTION_ERROR_INJECTION": null,
  "CONFIG_BPF_KPROBE_OVERRIDE": null,
  "CONFIG_NET": null,
  "CONFIG_XDP_SOCKETS": null,
  "CONFIG_LWTUNNEL_BPF": null,
  "CONFIG_NET_ACT_BPF": null,
  "CONFIG_NET_CLS_BPF": null,
  "CONFIG_NET_CLS_ACT": null,
  "CONFIG_NET_SCH_INGRESS": null,
  "CONFIG_XFRM": null,
  "CONFIG_IP_ROUTE_CLASSID": null,
  "CONFIG_IPV6_SEG6_BPF": null,
  "CONFIG_BPF_LIRC_MODE2": null,
  "CONFIG_BPF_STREAM_PARSER": null,
  "CONFIG_NETFILTER_XT_MATCH_BPF": null,
  "CONFIG_BPFILTER": null,
  "CONFIG_BPFILTER_UMH": null,
  "CONFIG_TEST_BPF": null
}
```

However, without the option `-j`, it shows the info:

```
❯ sudo ./bpftool feature probe
Scanning system configuration...
bpf() syscall restricted to privileged users
JIT compiler is enabled
JIT compiler hardening is disabled
JIT compiler kallsyms exports are disabled
Global memory limit for JIT compiler for unprivileged users is 264241152 bytes
skipping kernel config, can't open file: No such file or directory
[...]
```

But I don't think we want to run bpftool without `-j` just to catch that line. :/

And to be honest, I don't have any solution in my mind which is not ugly. Maybe we should just make a change in bpftool which will show the warning about lack of the kernel config also in JSON object when `-j` is used? I can do that.

Reply:
OK, yes, we can check for those files in Cilium too.

Reply:
Yeah, I was about to suggest to replicate bpftool's logic here to check for config paths.